awx/ui_next/src/screens/Job/JobOutput/JobOutput.jsx

```diff
@@ -108,9 +152,32 @@ class JobOutput extends Component {
   }
 
   componentWillUnmount() {
+    if (ws) {
+      ws.close();
+    }
+    clearInterval(this.interval);
     this._isMounted = false;
   }
 
+  monitorJobSocketEvents() {
+    const { remoteRowCount } = this.state;
+    let highestCounter = 0;
+    for (const jobEvent of this.jobSocketEvents) {
+      if (jobEvent.counter && jobEvent.counter > highestCounter) {
+        highestCounter = jobEvent.counter;
+      } else if (jobEvent.final_counter) {
+        highestCounter = jobEvent.final_counter;
+      }
+    }
+    if (highestCounter >= remoteRowCount) {
+      this._isMounted &&
+        this.setState({
+          remoteRowCount: highestCounter + 1,
+        });
+    }
+    this.jobSocketEvents.length = 0;
+  }
```

Comment:
All we actually need from the websocket events is to look at their counters so we can keep track of the total number of events expected to exist on the backend.

A next step might be to put the job events in a store that we check prior to making an http request for more rows when the user scrolls somewhere new. But it's not strictly needed.